cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/condition/ConditionContext.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.etl.api.condition;
+
+import co.cask.cdap.api.Transactional;
+import co.cask.cdap.api.security.store.SecureStore;
+import co.cask.cdap.api.security.store.SecureStoreManager;
+import co.cask.cdap.etl.api.StageContext;
+
+/**
+ * Represents the context available to the condition plugin during runtime.
+ */
+public interface ConditionContext extends StageContext, Transactional, SecureStore, SecureStoreManager {
+  /**
+   * Return the statistics (such as number of input records, output records, error records etc.) associated with the
+   * stages in the pipeline.
```

Comment:
Yeah it make sense to return the `Map<String, StageStatistics>` from `ConditionContext`
in that case we will not need to worry about whether stage is executed or not.. as this map will only have for the stages which are executed and then we can return 0 instead of `null` from StageStatistics object. will fix code to address this.